centralized-sampling-tests/sample-apps/golang-http-server/application.go

```diff
@@ -0,0 +1,160 @@
+)
+
+func getSampledSpanCount(name string, totalSpans string, attributes []attribute.KeyValue) int {
+	tracer := otel.Tracer(name)
+
+	var sampleCount = 0
+	totalSamples, err := strconv.Atoi(totalSpans)
+	if err != nil {
+		log.Println(err)
+		return -1
+	}
+
+	ctx := context.Background()
+
+	for i := 0; i < totalSamples; i++ {
+		_, span := tracer.Start(ctx, name, oteltrace.WithSpanKind(oteltrace.SpanKindServer), oteltrace.WithAttributes(attributes...))
+
+		if span.SpanContext().IsSampled() {
+			sampleCount++
+		}
+
+		span.End()
+	}
+
+	return sampleCount
+}
+
+func webServer() {
+	http.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("healthcheck"))
+		if err != nil {
+			log.Println(err)
+		}
+	}))
+
+	http.Handle("/getSampled", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		userAttribute := r.Header.Get("User")
+		required := r.Header.Get("Required")
+		serviceName := r.Header.Get("Service_name")
+		totalSpans := r.Header.Get("Totalspans")
+
+		var attributes = []attribute.KeyValue{
+			attribute.KeyValue{"http.method", attribute.StringValue(r.Method)},
+			attribute.KeyValue{"http.url", attribute.StringValue("http://localhost:8080/getSampled")},
+			attribute.KeyValue{"user", attribute.StringValue(userAttribute)},
+			attribute.KeyValue{"http.route", attribute.StringValue("/getSampled")},
+			attribute.KeyValue{"required", attribute.StringValue(required)},
+			attribute.KeyValue{"http.target", attribute.StringValue("/getSampled")},
+		}
+
+		var totalSampled = getSampledSpanCount(serviceName, totalSpans, attributes)
+		_, err := w.Write([]byte(strconv.Itoa(totalSampled)))
+		if err != nil {
+			log.Println(err)
+		}
+	}))
+
+	http.Handle("/importantEndpoint", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		userAttribute := r.Header.Get("User")
+		required := r.Header.Get("Required")
+		serviceName := r.Header.Get("Service_name")
+		totalSpans := r.Header.Get("Totalspans")
+
+		var attributes = []attribute.KeyValue{
+			attribute.KeyValue{"http.method", attribute.StringValue("GET")},
+			attribute.KeyValue{"http.url", attribute.StringValue("http://localhost:8080/importantEndpoint")},
+			attribute.KeyValue{"user", attribute.StringValue(userAttribute)},
+			attribute.KeyValue{"http.route", attribute.StringValue("/importantEndpoint")},
+			attribute.KeyValue{"required", attribute.StringValue(required)},
+			attribute.KeyValue{"http.target", attribute.StringValue("/importantEndpoint")},
+		}
+
+		var totalSampled = getSampledSpanCount(serviceName, totalSpans, attributes)
+		_, err := w.Write([]byte(strconv.Itoa(totalSampled)))
+		if err != nil {
+			log.Println(err)
+		}
+	}))
+
+	listenAddress := os.Getenv("LISTEN_ADDRESS")
+	if listenAddress == "" {
+		listenAddress = "localhost:8080"
+	}
+	log.Println("App is listening on %s !", listenAddress)
+	_ = http.ListenAndServe(listenAddress, nil)
+}
+
+func start_xray() (bool, error) {
```

Comment:
```suggestion
func start_xray()  error {
```
There is no reason to return a the boolean here since you are not doing anything with it.